upup/pkg/fi/cloudup/bootstrapchannelbuilder.go

```diff
@@ -381,3 +496,47 @@ func (b *BootstrapChannelBuilder) buildManifest() (*channelsapi.Addons, map[stri
 
 	return addons, manifests, nil
 }
+
+func BuildSecret() (kube_api.Secret, error) {
```

Comment:
Does this need to be public? Is there code upstream that we can use rather than maintaining our own?